src/reducers/comments.spec.tsx

```diff
@@ -112,37 +123,106 @@ describe(__filename, () => {
       );
       state = reducer(state, actions.finishComment(keyParams));
 
-      expect(state.byKey[createCommentKey(keyParams)]).toMatchObject({
+      expect(
+        selectCommentInfo({ comments: state, ...keyParams }),
+      ).toMatchObject({
         pendingCommentText: null,
         savingComment: false,
       });
     });
   });
 
-  describe('setComment', () => {
-    it('sets a comment for a key', () => {
-      const comment = createFakeExternalComment({ id: 54321 });
+  describe('setComments', () => {
+    it('sets comments for a key', () => {
+      const comment1 = createFakeExternalComment({ id: 54321 });
+      const comment2 = createFakeExternalComment({ id: 54322 });
+
       const versionId = 1;
       const line = 123;
       const fileName = 'manifest.json';
 
       const state = reducer(
         undefined,
-        actions.setComment({ comment, fileName, line, versionId }),
+        actions.setComments({
+          comments: [comment1, comment2],
+          fileName,
+          line,
+          versionId,
+        }),
       );
 
-      expect(state.byId[comment.id]).toEqual(createInternalComment(comment));
+      expect(state.byId[comment1.id]).toEqual(createInternalComment(comment1));
+      expect(state.byId[comment2.id]).toEqual(createInternalComment(comment2));
 
       expect(
-        state.byKey[createCommentKey({ fileName, line, versionId })],
+        selectCommentInfo({ comments: state, fileName, line, versionId }),
       ).toMatchObject({
-        commentIds: [comment.id],
+        commentIds: [comment1.id, comment2.id],
       });
+
+      expect(state.hasComments).toEqual(true);
+      expect(state.forVersionId).toEqual(versionId);
+    });
+
+    it('sets zero comments', () => {
+      const versionId = 1;
+      const state = reducer(
+        undefined,
+        actions.setComments({ comments: [], ...keyParams, versionId }),
+      );
+
+      expect(state.hasComments).toEqual(false);
+      expect(state.byId).toEqual({});
+      expect(state.forVersionId).toEqual(versionId);
+    });
+
+    it('preserves hasComments if the last state had comments', () => {
```

Comment:
I think you also need a test where the last state had 0 comments and we're setting a number of comments via the action. As I commented in the reducer, I don't think this case is handled in the reducer and I think such a test would fail.